rizinorg/cutter
# Start making adjustment for qt6 compatibility.

**Your checklist for this pull request**


**Detailed description**

Looks like even rolling release distros might keep qt6 and qt5 available for some time so there is no rush.

This PR introduces option for attempting to compile with qt6. It doesn't compile yet, biggest changes remaining are model filter regexps and QSVgWidget.  Tried building with qt6 on Archlinux, it has [qt6](https://www.archlinux.org/packages/?q=qt6) beta packages which can be installed in parallel with qt5.

**Test plan (required)**

* CI tests building on Ubuntu 16.04 against system Qt (5.5) compiles :heavy_check_mark: 
* Make sure resizing behavior for combo boxes in hexwidget side panel is reasonable :heavy_check_mark: 
* Test that dock locking unlocking works
  * old settings upgrade - open old cutter to trigger saving settings, open the new cutter and make sure restored dock locking state matches the old cutter, try with both states :heavy_check_mark: 
  * clean state with settings deleted :heavy_check_mark: 
  * toggle and make sure effect is correct :heavy_check_mark: 
  * save and reopen with both states :heavy_check_mark: 

**Related issues**

#2464
